apps/web-remix/app/routes/forms.$organizationId.pipelines.$pipelineId.tsx

```
@@ -33,18 +59,91 @@ export async function loader(args: LoaderFunctionArgs) {
 export default function WebsiteForm() {
   const { pipelineId, organizationId, pipeline } =
     useLoaderData<typeof loader>();
-  const [outputs, setOutputs] = useState<Record<string, string>>({});
-  const [files, setFiles] = useState<Record<string, File>>({});
 
-  const validator = useMemo(() => withZod(z.any()), []);
+  const defaultOutputs = useCallback(
+    () =>
+      pipeline.interface_config.form.outputs.reduce(
+        (acc, curr) => ({
+          ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)